File: src/pocketcloud/cloud/network/packet/impl/normal/TemplateSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;
use pocketcloud\cloud\template\Template;

class TemplateSyncPacket extends CloudPacket {

    public function __construct(
        private ?Template $template = null,
        private bool $removal = false
    ) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeTemplate($this->template);
        $packetData->write($this->removal);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->template = $packetData->readTemplate();
        $this->removal = $packetData->readBool();
    }

    public function getTemplate(): ?Template {
        return $this->template;
    }

    public function isRemoval(): bool {
        return $this->removal;
    }

    public function handle(ServerClient $client): void {}

    public static function create(Template $template, bool $removal): self {
        return new self($template, $removal);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/TemplateSyncPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.model.template.ITemplate;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.common.mapper.MapperUtils;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class TemplateSyncPacket extends CloudPacket implements ClientboundPacket {

    private ITemplate template;
    private boolean removal;

    public TemplateSyncPacket(ITemplate template, boolean removal) {
        this.template = template;
        this.removal = removal;
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(MapperUtils.toMap(template), removal);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static TemplateSyncPacket create(ITemplate template, boolean removal) {
        return new TemplateSyncPacket(template, removal);
    }
}
